go/workflow.go

```diff
@@ -0,0 +1,333 @@
+package loyalty
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+const TaskQueue = "CustomerLoyaltyTaskQueue"
+const EventsThreshold = 10000
+
+// Signal, query, and error string constants
+const (
+	SignalCancelAccount       = "cancelAccount"
+	SignalAddPoints           = "addLoyaltyPoints"
+	SignalInviteGuest         = "inviteGuest"
+	SignalEnsureMinimumStatus = "ensureMinimumStatus"
+	QueryGetStatus            = "getStatus"
+	QueryGetGuests            = "getGuests"
+)
+
+const (
+	EmailWelcome            = "Welcome to our loyalty program! You're starting out at '%v' status."
+	EmailGuestCanceled      = "Sorry, your guest has already canceled their account."
+	EmailGuestInvited       = "Congratulations! Your guest has been invited!"
+	EmailInsufficientPoints = "Sorry, you need to earn more points to invite more guests!"
+	EmailPromoted           = "Congratulations! You've been promoted to '%v' status!"
+	EmailDemoted            = "Unfortunately, you've lost enough points to bump you down to '%v' status. 😞"
+	EmailCancelAccount      = "Sorry to see you go!"
+)
```

Comment:
Questionable benefit to moving these format strings away from where they are used (it's often hard for readers/devs to know the params a phrase may need if the phrase is somewhere else, much less keep them in sync). But if you must put them up here, might as well not expose them (i.e. do not capitalize).

Reply:
I've at least not exposed them now. I want to keep them as consts here as I use them in tests to check if the right emails are [attempted to be] sent at the right time(s).